### PR TITLE
added support for paginated requests

### PR DESCRIPTION
**Pull request by dpawluk**
also fixed up some request formatting, zat validates, all good things.

@maximeprades if you want to CC someone to +1 not sure who owns this.

### CODE REVIEW

**Comment by maximeprades** [comment]
cc @zendesk/quokka

**Comment by adammw** [comment]
Tracking this as https://zendesk.atlassian.net/browse/OFFAPPS-451 / https://support.zendesk.com/agent/tickets/1031624.

File: app.js [hunk 1]
```diff
@@ -311,7 +322,39 @@
 
     onTokenDelete: function(e) {
       this.$(e.target).parent('li.token').remove();
-    }
+    },
+
+    _paginate: function(a) {
```
Comment (adammw) [inline app.js]: I found the `_paginate` method rather hard to understand, not exactly sure how it works.

Comment (dpawluk) [inline app.js]: I'm making another pull with a better function and line by line notes @adammw 
I didn't find a much better way to do it but I removed a lot of the uglier underscore but it is still stupidly ugly.

**Comment by dpawluk** [comment]
I'm not sure if closing this will do anything inre:tracking so I will leave that to anyone else. I made a new pull with a cleaner more annotated pagination f(x)

https://github.com/zendesk/notification_app/pull/33

**Comment by princemaple** [comment]
closed in favor of #33

**Pull request closed without merging**